src/libraries/System.Data.OleDb/src/OleDbConnectionFactory.cs

```diff
@@ -33,8 +33,7 @@ public override DbProviderFactory ProviderFactory
 
         protected override DbConnectionInternal CreateConnection(DbConnectionOptions options, DbConnectionPoolKey poolKey, object poolGroupProviderInfo, DbConnectionPool? pool, DbConnection? owningObject)
         {
-            // TODO-NULLABLE: owningObject may actually be null (see DbConnectionPool.CreateObject), in which case this will throw...
-            DbConnectionInternal result = new OleDbConnectionInternal((OleDbConnectionString)options, (OleDbConnection)owningObject!);
```

Comment:
the comment seemed bogus as the input argument is actually nullable and I double checked and it handles null just fine